### PR TITLE
ENH added first pass at classes for coadds and sources

### PR DESCRIPTION
**Pull request by beckermr**
This PR pulls out the relevant code for dealing with DESDM.

### CODE REVIEW

File: pizza_cutter/des_coadd_data.py [hunk 1]
```diff
@@ -0,0 +1,653 @@
...
+_DOWNLOAD_CMD = r"""
+    rsync \
+        -av \
+        --password-file $DES_RSYNC_PASSFILE \
+        --files-from=%(flist_file)s \
+        %(userstring)s${DESREMOTE_RSYNC}/ \
+        %(source_dir)s/
+"""
+
+_QUERY_COADD_TEMPLATE_BYTILE_DESSCI = """\
```
Comment (beckermr) [inline pizza_cutter/des_coadd_data.py]: @esheldon Can you look at this SQL? It appears to work, but you know the database schema's better than I do.

Comment (esheldon) [reply pizza_cutter/des_coadd_data.py]: It used to work I think.  It depends on which database you are using probably.  Some queries only work in desoper

Comment (beckermr) [reply pizza_cutter/des_coadd_data.py]: Right. It works when I use it against dessci. I just want to make sure the query is correct. Completion is a low bar.

Comment (esheldon) [reply pizza_cutter/des_coadd_data.py]: We can check against the file lists in existing MEDS files

Comment (beckermr) [reply pizza_cutter/des_coadd_data.py]: Good idea.

Comment (beckermr) [reply pizza_cutter/des_coadd_data.py]: Closing the loop here. I am getting this path

`OPS/finalcut/Y2A1/Y1-2363/20131010/D00242799/p01/`

from my DB queries on `desoper` but the existing meds file has

`OPS/finalcut/Y2A1/Y1-2363/20131010/D00242799/p06/`

IDK what the difference is. Any ideas @esheldon?

Comment (beckermr) [reply pizza_cutter/des_coadd_data.py]: Ahhhhh. This is Y3A1_COADD versus Y3A2_COADD. My tests use Y3A1_COADD but I downloaded the Y3A2_COADD. Comparing to the right MEDS file has the right paths. Merging.

**Comment by esheldon** [comment]
I think all the tiles are also part of `y3a2_coadd` tag possibly, but I specifically did not take everything from there because I had already begun processing with `y3a1_coadd`, and only after did the reprocessed subset get created

**Pull request merged**